design-proposals/persist-vmi-firmware-uuid.md

```diff
@@ -0,0 +1,167 @@
+
+Users would be notified of this change in advance through logs or alerts,
+and optional tooling could be provided to help users preserve their current firmware UUIDs by adding them to the spec.
+
+**Pros:**
+- Ensures universally unique UUIDs, meeting the UUID standard and avoiding conflicts when migrating VMs across clusters.
+- Keeps the spec clean without repurposing fields.
+- Simple to implement without introducing new API fields.
+
+**Cons:**
+- This approach is a breaking change, requiring user intervention to preserve existing UUIDs.
+- Requires additional communication and support, as users may need to update or reconfigure their VMs to prevent disruptions.
+- May necessitate tooling to facilitate UUID preservation for compatibility with existing workloads.
+
+---
+
+### 4. Upgrade-Specific Persistence of Firmware UUID
+
+**Description:** Before upgrading KubeVirt, persist the `Firmware.UUID` of existing VMsin `vm.spec`.
+After the upgrade, any VM without `Firmware.UUID` is considered new.
+For these new VMs, use `vm.metadata.uid` as the firmware UUID if `vm.spec.template.spec.Firmware.UUID` is not defined.
+
+**Pros:**
+- Upgrade-specific code can be removed after two or three releases.
+- Maintains a straightforward logic post-upgrade, with minimal changes required.
+- Limits disturbance to GitOps workflows to pre-existing VMs that had potentially buggy UUID behavior.
+
+**Cons:**
+- Requires additional code (likely in `virt-operator`) to handle the upgrade-specific persistence.
+- Code should be added to the computation of the "restart required" condition, so that it is not raised if vm.Template.Firmware.UUID equals vmi.Firmware.UUID.
+
+---
+
+## What Happens to the Firmware UUID During a Restore?
+* The firmware UUID is part of the VM's spec, or dynamically generated by the VM controller if it is not defined.
+* Snapshots created before implementing the firmware UUID persistence mechanism will not include this field.
+* As a result, during a restore, a new UUID may be generated if the vm.spec.template.spec.firmware.uuid field is absent.
+
+**Note:** Focusing on persistence ensures that existing behavior remains unchanged, avoiding restore issues while laying a foundation for future UUID management enhancements.
+
+
+## API Examples
+**Example uuid persistence within VM Status**
+
+```yaml
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: mytestvm
+status:
+   conditions:
+      - lastProbeTime: "2024-11-06T01:12:29Z"
+        lastTransitionTime: "2024-11-06T01:12:29Z"
+        message: Guest VM is not reported as running
+        reason: GuestNotRunning
+        status: "False"
+        type: Ready
+   created: true
+   runStrategy: Once
+   firmwareUUID: "123e4567-e89b-12d3-a456-426614174000"
+```
+
+## Scalability
+The proposed changes have no anticipated impact on scalability capabilities of the KubeVirt framework
+
+## Update/Rollback Compatibility
+should not affect updates / rollbacks.
```

Comment:
This is where the content of `## What Happens to the Firmware UUID During a Restore?` should land.
Or you should remove this part.